src/abuf.h

```diff
@@ -35,13 +35,17 @@
 #ifndef AXEL_ABUF_H
 #define AXEL_ABUF_H
 
+#define ABUF_CHUNK 512
+
 typedef struct {
 	char *p;
+	size_t pos; /* end of string */
```

Comment:
To add `pos` you need to implement it everywhere, and initialize it at `abuf_setup`. However, I recommend against using this sort of thing unless you can prove that:
- it's actually more efficient than letting the allocator decide
- you can keep memory growth under control

In the end to solve those constraints alone you would need to implement another allocator...

I'm not saying it's a bad idea to do so, it's just the wrong place.

Reply:
My original intention of using this variable is to replace strlcat with abuf_append so that it is not necessary to find the end of string every time. But now I have some doubts whether such optimization is necessary...

Reply:
well, the buffers don't shrink, so if you want to optimize append that way you need to allow shrinking or keep track of the zero, which would be valid usage, but I think it may not be worth the optimization. The append function abstraction however is fine.

Reply:
Ok, I will only do the abstraction.